roofit/hs3/src/JSONFactories_HistFactory.cxx

```diff
@@ -366,37 +376,33 @@ bool importHistSample(RooJSONFactoryWSTool &tool, RooDataHist &dh, Scope &scope,
                                                            : "alpha_" + sysname);
             RooAbsReal *par = ::getNP(ws, parname);
             histo_nps.add(*par);
-            RooArgSet varlist;
-            scope.getObservables(varlist);
-            auto dh_low = RooJSONFactoryWSTool::readBinnedData(p["data"], sysname + "Low_" + nameWithPrefix, varlist);
-            ownedArgsStack.push(makeUnique<RooHistFunc>(sysname + "Low_" + nameWithPrefix, *dh_low->get(), *dh_low));
-            histo_low.add(*ownedArgsStack.top());
-            auto dh_high = RooJSONFactoryWSTool::readBinnedData(p["data"], sysname + "High_" + nameWithPrefix, varlist);
-            ownedArgsStack.push(makeUnique<RooHistFunc>(sysname + "High_" + nameWithPrefix, *dh_high->get(), *dh_high));
-            histo_high.add(*ownedArgsStack.top());
+	    auto &data = mod["data"];
+            ownedHistStack.push(std::move(RooJSONFactoryWSTool::readBinnedData(data["lo"], sysname + "Low_" + nameWithPrefix, varlist)));
+	    ownedArgsStack.push(makeUnique<RooHistFunc>(sysname + "Low_" + nameWithPrefix, varlist, *ownedHistStack.top()));
```

Comment:
```suggestion
	    ownedArgsStack.push(makeUnique<RooHistFunc>(sysname + "Low_" + nameWithPrefix, varlist, RooJSONFactoryWSTool::readBinnedData(data["lo"], sysname + "Low_" + nameWithPrefix, varlist)));
```

The `RooHistFunc` has a constructor that directly takes `std::unique_ptr<RooDataHist>` to take the ownership of it. Then you don't need to track the ownership separately.

Before doing your further validation tomorrow, can you maybe change this? Also a few lines below. Thanks!

Reply:
fixed.